### PR TITLE
ART-7954 scan-sources:noop - Fetch commit for the metadata file

### PR DESCRIPTION
**Pull request by thegreyd**
Follow up to https://github.com/openshift-eng/art-tools/pull/110

Test
- File that has noop commit - `./doozer --assembly=stream --group 'openshift-4.15'@4.15-update-build-data-org --data-path https://github.com/locriandev/ocp-build-data --images atomic-openshift-cluster-autoscaler config:scan-sources`
- File without noop commit - `./doozer --assembly=stream --group 'openshift-4.15'@4.15-update-build-data-org --data-path https://github.com/locriandev/ocp-build-data --images openshift-base-nodejs config:scan-sources`

### CODE REVIEW

File: doozer/doozerlib/cli/scan_sources.py [hunk 1]
```diff
@@ -198,14 +198,17 @@ def check_config_changes(self, image_meta, build_info):
...
+                    path = f'images/{image_meta.config_filename}'
+                    rc, commit_message, _ = exectools.cmd_gather(f'git log -1 --format=%s -- {path}', strip=True)
                     if rc != 0:
-                        raise IOError(f'Unable to retrieve commit message from {self.runtime.data_dir}')
+                        raise IOError(f'Unable to retrieve commit message from {self.runtime.data_dir} for {path}')
 
                 if commit_message.lower().startswith('scan-sources:noop'):
-                    self.runtime.logger.info('Ignoring digest change since commit message indicates noop')
+                    self.runtime.logger.info('Digest change not considered for rebuild since commit message indicates noop')
+                    self.add_assessment_reason(image_meta, RebuildHint(RebuildHintCode.NOOP_CONFIG_CHANGE,
```
Comment (locriandev) [inline doozer/doozerlib/cli/scan_sources.py:210]: Why is this needed?

Comment (thegreyd) [reply doozer/doozerlib/cli/scan_sources.py:210]: Not necessary, but added it for visibility, we'll get to see this reason in the report that's generated at the end

Comment (thegreyd) [reply doozer/doozerlib/cli/scan_sources.py]: Removed

**Review by locriandev — COMMENTED** [review]
/lgtm
